internal/queryRunner.go

```diff
@@ -170,3 +185,47 @@ func (queryRunner *PgQueryRunner) stopBackup() (label string, offsetMap string,
 
 	return label, offsetMap, lsnStr, nil
 }
+
+// GetParameter reads a Postgres setting
+// TODO: Unittest
+func (queryRunner *PgQueryRunner) GetParameter(parameterName string) (string, error) {
+	var value string
+	conn := queryRunner.connection
+	err := conn.QueryRow(queryRunner.buildGetParameter(), parameterName).Scan(&value)
+	return value, err
+}
+
+// GetWalSegmentBytes reads the wals segment size (in bytes) and converts it to uint64
+// TODO: Unittest
+func (queryRunner *PgQueryRunner) GetWalSegmentBytes() (segBlocks uint64, err error) {
+	strValue, err := queryRunner.GetParameter("wal_segment_size")
+	if err != nil {
+		return 0, err
+	}
+	segBlocks, err = strconv.ParseUint(strValue, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	if queryRunner.Version <= 100000 {
+		// For PG 10 and below, wal_segment_size is in 8k blocks
+		segBlocks *= 8192
```

Comment:
FWIW we currently only support 16Mb segments.... But I think this can be configured in future

Reply:
Correct, I think it will work with custom segment sizes, but I have not tested it yet.